File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int num_letters = count_letters(text);
    int num_words = count_words(text);
    int num_sentences = count_sentences(text);
    float L = (float) num_letters / num_words * 100;
    float S = (float) num_sentences / num_words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

// count the amount of alphabetical characters from text
int count_letters(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }

    return count;
}

// count the amount of words separated with a space in text
int count_words(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    }
    // increment count one extra time for the last word
    count++;

    return count;
}


// count the amount of sentences in text
int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char ch = text[i];
        if (ch == '.' || ch == '!' || ch == '?')
        {
            count++;
        }
    }

    return count;
}

File: pset6/readability/readability.py
import cs50
import re


def main():
    text = cs50.get_string("Text: ")
    num_letters = (count_letters(text))
    num_words = (count_words(text))
    num_sentences = (count_sentences(text))

    L = num_letters / num_words * 100
    S = num_sentences / num_words * 100
    index = 0.0588 * L - 0.296 * S - 15.8

    # print(f"Letters: {num_letters}")
    # print(f"Words: {num_words}")
    # print(f"Sentences: {num_sentences}")
    # print(f"L {L}")
    # print(f"S {S}")
    # print(f"index {index}")

    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")

    
def count_letters(text):
    count = 0
    for ch in text:
        if ch.isalpha():
            count += 1
    
    return count


def count_words(text):
    return len(text.split(" "))


def count_sentences(text):
    count = 0
    for ch in text:
        if ch == "." or ch == "!" or ch == "?":
            count += 1

    return count


main()